test/spmd/tensor/test_matrix_ops.py

```diff
@@ -127,6 +129,111 @@ def test_t(self):
         self.assertEqual(tranposed_mat2.size(), torch.Size([12, 8]))
         self.assertEqual(tranposed_mat2.placements, shard_spec)
 
+    # TODO: Need to investigate why test failed in CPU for baddbmm.
+    @with_comms
+    @skip_if_lt_x_gpu(TEST_GPU_NUM)
+    def test_sharded_baddbmm(self):
+        # If beta is 0, input tensor will be ignored
+        device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+        torch.manual_seed(self.rank)
+        tensor = torch.rand(3, 5, 6, device=self.device_type)
+        batch_1 = torch.rand(3, 5, 8, device=self.device_type)
+        batch_2 = torch.rand(3, 8, 6, device=self.device_type)
+        local_result = torch.baddbmm(
+            tensor, batch_1, batch_2, beta=0.0, alpha=0.5
+        )
+        shard0_spec = [Shard(0)]
+        shard1_spec = [Shard(1)]
+        shard2_spec = [Shard(2)]
+
+        def test_comb(
+            tensor_placements: Sequence[Placement],
+            batch_1_placements: Sequence[Placement],
+            batch_2_placements: Sequence[Placement],
+            beta: int,
+            alpha: int,
+            sharding_dim: int
+        ) -> None:
+            tensor_dt = DTensor.from_local(tensor, device_mesh, tensor_placements)
+            batch_1_dt = DTensor.from_local(batch_1, device_mesh, batch_1_placements)
+            batch_2_dt = DTensor.from_local(batch_2, device_mesh, batch_2_placements)
+            new_dt = torch.baddbmm(
+                tensor_dt, batch_1_dt, batch_2_dt, beta=0.0, alpha=0.5
+            )
+            self.assertTrue(new_dt.placements[0].is_shard(dim=sharding_dim))
+            self.assertEqual(new_dt.to_local(), local_result)
+
+        test_comb(
+            shard0_spec, shard0_spec, shard0_spec,
+            beta=0.8, alpha=0.5, sharding_dim=0
+        )
+
+        test_comb(
+            shard2_spec, shard2_spec, shard2_spec,
+            beta=0.0, alpha=0.5, sharding_dim=0
+        )
+
+        # sharding on non-batching dim raises exception:
+        # Two different input dims cannot sharded across the same mesh dim
+        with self.assertRaises(Exception):
```

Comment:
shall we group all those checks together? I guess sth like:
tensor_spec = [shard0/1/2]
batch1_spec = [shard0/1/2]
batch2_spec = [shard0/1/2]

combs = list(itertools.product(tensor_spec, batch1_spec, batch2_spec))
then remove the ones that get passed already (or have a pass list), and assert the rest of them having exceptions?

Reply:
btw you could just use DTensorConverter to automatically generate all combinations for every argument https://github.com/pytorch/tau/blob/c4158aa580f982517277e21e749f549d100c8952/spmd/test/common_utils.py#L96

Reply:
> shall we group all those checks together? I guess sth like: tensor_spec = [shard0/1/2] batch1_spec = [shard0/1/2] batch2_spec = [shard0/1/2]
> 
> combs = list(itertools.product(tensor_spec, batch1_spec, batch2_spec)) then remove the ones that get passed already (or have a pass list), and assert the rest of them having exceptions?

This part for `bmm` is done. Next is to add for `baddbmm` and test on GPU.



> btw you could just use DTensorConverter to automatically generate all combinations for every argument
> 
> https://github.com/pytorch/tau/blob/c4158aa580f982517277e21e749f549d100c8952/spmd/test/common_utils.py#L96

Need to see how to use it. But my current implementation should be able to port to 3 operands case (i.e. `baddbmm`) easily.